webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Details/Cards/CloudBilling/index.js

```diff
@@ -0,0 +1,123 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { translate as __ } from '../../../../../../common/I18n';
+import CardTemplate from '../../../../Templates/CardItem/CardTemplate';
+import BillingAws from './BillingAws';
+import BillingAzure from './BillingAzure';
+import BillingGcp from './BillingGcp';
+
+// Helper function to detect provider from billing facts
+// Uses representative facts to determine the cloud provider
+const detectProviderFromFacts = data => {
+  if (!data) return null;
+
+  /* eslint-disable spellcheck/spell-checker */
```

Comment:
Now that there's the provider value, can this be a simple case on that value instead of a complex if tree?